hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/access/ReadWriteAccessDelegate.java

```diff
@@ -57,6 +80,29 @@ public void unlockItem(final Object key, final SoftLock lock) throws CacheExcept
         cache.unlock(key, lock);
     }
 
-    public void unlockRegion(SoftLock lock) throws CacheException {
+    /**
+     * {@inheritDoc}
+     * <p/>
+     * A no-op since this is an asynchronous cache access strategy.
+     */
+    @Override
+    public boolean update(Object key, Object value, Object currentVersion, Object previousVersion)
```

Comment:
Before this will be able to be sent to upstream we'll need to go back over it to `final` all the things, switch wildcard imports back to explicit, etc., to meet their code standards. Just noting it